power_grid_model_c/power_grid_model/include/power_grid_model/common/three_phase_tensor.hpp

```diff
@@ -351,13 +351,14 @@ template <symmetry_tag sym, class Proxy>
 inline void update_real_value(RealValue<sym> const& new_value, Proxy&& current_value, double scalar) {
     if constexpr (is_symmetric_v<sym>) {
         if (!is_nan(new_value)) {
-            current_value = scalar * new_value;
+            std::forward<Proxy>(current_value) = scalar * new_value;
```

Comment:
also this seems to be very strange.

Reply:
this is how a perfectly forwarded proxy works: the `current_value` is either taken by `&` or by `&&` or by value depending on what the type in caller is, and then here it is called as

```cpp
std::forward<Proxy>.operator=(scalar * new_value);
```
which can be rewritten as `std::forward<Proxy>(current_value) = scalar * new_value;`

This deliberately consumes the `current_value` in the case of forwarding on a prvalue.

If we do not want to do perfect forwarding, can you please explain why we went for this solution in the first place here? I do understand that this implementation creates a modifiable view of the 3-phase tensor by value, while taking mutable reference for other objects, but it this really the way that `Eigen` wants us to do it?

Reply:
see https://github.com/PowerGridModel/power-grid-model/pull/882#discussion_r1938977795